docs/EXTENDED-ARCHIVES.md

```diff
@@ -0,0 +1,413 @@
+compression format, which has the following benefits:
+
+- A native, robust, performant, cross-platform and streaming-based
+  decompression library for [node.js][nodejs], called
+  [`lzma-native`][lzma-native], which we battle tested at Etcher
+
+- A way to calculate the precise uncompressed size of the file no matter the
+  size (which is a [known limitation of `gzip` after 4
+  GB](https://github.com/resin-io/etcher/issues/638)), and an operation
+  [`lzma-native`][lzma-native] supports out of the box. This is not possible
+  with `bzip2`
+
+- Decent compression ratio, and compression/decompression time
+
+Archive layout
+--------------
+
+The contents of an extended archive are the followings:
+
+- **REQUIRED** `.meta/manifest.json`
+- `.meta/publisher.svg`
+- `.meta/device.svg`
+- `.meta/image.svg`
+- `.meta/schema.json`
+- `.meta/image.bmap`
+- `.meta/instructions.markdown`
+- `.meta/release-notes.txt`
+- **REQUIRED** `image.<extension>`
+
+### Order
+
+Notice that the order is important. The `.meta/` directory must come before the
+image data, since we need that information before being able to make use of the
+image.
+
+The order of the files inside `.meta/` is not mandatory, however the order
+described above ensures the maximum UX perceived efficiency.
+
+### Graphics
+
+We chose [`svg`][svg] over other image formats since clients can display it on
+any kind of screen resolution without requiring multi-layer files, or multiple
+files workarounds.
+
+Archive files
+-------------
+
+### `.meta/manifest.json`
+
+This is the only required file inside `.meta`, which declares information about
+the image, the target device type, and the publisher. This file may contain the
+following top level properties:
+
+- `publisher`: Information about the image publisher
+- `device`: Information about the image device type
+- `image`: Information about the image itself
+
+#### Publisher
+
+Here's an example of a real-world publisher manifest for RetroPie:
+
+```json
+{
+  "name": "RetroPie",
+  "url": "https://retropie.org.uk",
+  "supportUrl": "https://retropie.org.uk/forum/",
+  "colorScheme": {
+    "background": "#535760",
+    "text": "#FFFFFF",
+    "primary": "#5793db"
+  }
+}
+```
+
+Describe a publisher with any of the following properties:
+
+##### `name (String)`
+
+The display human-friendly name of the publisher.
+
+##### `url (String)`
+
+The main url of the publisher, usually the landing page.
+
+##### `supportUrl (String)`
+
+The publisher url where users can get general support.
+
+This could be a link to a forum, IRC room, troubleshooting page, support form,
+etc.
+
+##### `colorScheme (Object)`
+
+The publisher specific color scheme, if any.
+
+The client may allow a certain degree of branding based on this color scheme.
+
+You may declare the following colors, in hexadecimal format:
+
+- `background`: The background color
+- `text`: The text color
+- `primary`: The primary color
+
+#### Device
```

Comment:
I'm starting to think there should be a collection of device manifests somewhere centralized (e.g: in an Etcher server) for the sole reason of asking the user: What is your device type? And then allowing them to download and flash extended archives that we host that match that criteria.

In that case, we can leave out the device manifests from the extended archive, and only allow images to declare a list of device types slugs they support.

@alexandrosm Am I missing any use case we discussed at some point?